src/twisted/internet/base.py

```diff
@@ -587,32 +600,307 @@ class ReactorBase(PluggableResolverMixin):
         register the thread it is running in as the I/O thread when it starts.
         If C{True}, registration will be done, otherwise it will not be.
+
     @ivar _exitSignal: See L{_ISupportsExitSignalCapturing._exitSignal}
+
+    @ivar _installSignalHandlers: A flag which indicates whether any signal
+        handlers will be installed during startup.  This includes handlers for
+        SIGCHLD to monitor child processes, and SIGINT, SIGTERM, and SIGBREAK
+        to stop the reactor.
     """
 
     _registerAsIOThread = True
 
-    _stopped = True
     installed = False
     usingThreads = False
     _exitSignal = None
 
     __name__ = "twisted.internet.reactor"
 
-    def __init__(self) -> None:
+    # IReactorCore
+    @property
+    def running(self) -> bool:
+        return self._core.running
+
+    # IReactorPluggableNameResolver
+    @property
+    def nameResolver(self) -> IHostnameResolver:
+        return self._core.nameResolver
+
+    # Not part of IReactorPluggableResolver but some tests use it and it has
+    # no leading underscore ... :/
+    @property
+    def resolver(self) -> IResolverSimple:
+        return self._core.resolver
+
+    def _addInternalReader(self, reader: IReadDescriptor) -> None:
+        self._internalReaders.add(reader)
+        self.addReader(reader)
+
+    def _removeInternalReader(self, reader: IReadDescriptor) -> None:
+        self._internalReaders.remove(reader)
+        self.removeReader(reader)
+
+    def __init__(self, *, _coreFactory: Optional[_CoreFactory] = None) -> None:
         super().__init__()
         self.threadCallQueue: List[_ThreadCall] = []
-        self._eventTriggers: Dict[str, _ThreePhaseEvent] = {}
         self._pendingTimedCalls: List[DelayedCall] = []
         self._newTimedCalls: List[DelayedCall] = []
         self._cancellations = 0
-        self.running = False
-        self._started = False
-        self._justStopped = False
-        self._startedBefore = False
+
+        coreFactory = _coreFactory if _coreFactory is not None else ReactorCore
+
+        # IReactorCore
+        self._core = coreFactory(
+            runUntilCurrent=self.runUntilCurrent,
+            doIteration=self.doIteration,
+            timeout=self.timeout,
+            wakeUp=self.wakeUp,
+        )
+        self.resolve = self._core.resolve  # type: ignore[assignment]
+        self.fireSystemEvent = self._core.fireSystemEvent  # type: ignore[assignment]
+        self.addSystemEventTrigger = self._core.addSystemEventTrigger  # type: ignore[assignment]
+        self.callWhenRunning = self._core.callWhenRunning  # type: ignore[assignment]
+        self.removeSystemEventTrigger = self._core.removeSystemEventTrigger  # type: ignore[assignment]
+        self.stop = self._core.stop  # type: ignore[assignment]
+        self.crash = self._core.crash  # type: ignore[assignment]
+        self.iterate = self._core.iterate  # type: ignore[assignment]
```

Comment:
I think this whole block is pretty ugly.  We need to keep exposing the `IReactorCore` interface here.  Instead of copying all these attributes over, we could use `proxyForInterface` to automate the delegation.  `proxyForInterface` results in a dynamic base class so mypy will never like it.  I think pydoctor is also a consideration.  All of these `type: ignore` comments are required because there are also stub implementations of these methods below, present only for doc generation.

I'm not sure which of these I like least ... maybe the one that's checked in though.